kv/cachedb.go

```diff
@@ -41,13 +41,13 @@ func (c *cachedb) Set(tableID int64, key Key, value []byte) error {
 		table = newMemDB()
 		c.memTables[tableID] = table
 	}
-
 	err := table.Set(key, value)
-	if err != nil {
-		return err
+	if ErrTxnTooLarge.Equal(err) {
```

Comment:
The function call to ErrTxnTooLarge is an expensive operation, could you check if err is nil first?

Reply:
Thanks, PTAL again.